components/accelerator/nvidia/query/infiniband/parse.go

```diff
@@ -0,0 +1,164 @@
+	cnt := 0
+	for _, card := range cards {
+		if card.Port1.Rate < rate {
+			continue
+		}
+
+		// e.g.,
+		// State: Active
+		// State: Down
+		if expectedState != "" && card.Port1.State != expectedState {
+			continue
+		}
+
+		// e.g.,
+		// Physical state: LinkUp
+		// Physical state: Disabled
+		if expectedPhysicalState != "" && card.Port1.PhysicalState != expectedPhysicalState {
+			continue
+		}
+
+		cnt++
+	}
+	return cnt
+}
+
+type IBStatCard struct {
+	Name            string     `json:"CA name"`
+	Type            string     `json:"CA type"`
+	NumPorts        string     `json:"Number of ports"`
+	FirmwareVersion string     `json:"Firmware version"`
+	HardwareVersion string     `json:"Hardware version"`
+	NodeGUID        string     `json:"Node GUID"`
+	SystemImageGUID string     `json:"System image GUID"`
+	Port1           IBStatPort `json:"Port 1"`
+}
+
+type IBStatPort struct {
+	State         string `json:"State"`
+	PhysicalState string `json:"Physical state"`
+	Rate          int    `json:"Rate"`
+	BaseLid       int    `json:"Base lid"`
+	LinkLayer     string `json:"Link layer"`
+}
+
+// ParseIBStat parses ibstat output and returns YAML representation
+func ParseIBStat(input string) (IBStatCards, error) {
+	scanner := bufio.NewScanner(strings.NewReader(input))
+
+	lines := make([]string, 0)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if strings.TrimSpace(line) == "" {
+			continue
+		}
+
+		// CA 'mlx5_9'
+		// should be
+		// - CA name: mlx5_9
+		// with the indentation
+		// but we convert them at the end
+		if strings.HasPrefix(line, "CA '") && !strings.HasPrefix(strings.TrimSpace(line), "CA type:") {
+			line = strings.TrimSpace(line)
+			line = strings.TrimPrefix(line, "CA '")
+			line = strings.TrimSuffix(line, "'")
+			lines = append(lines, "- CA name: "+line)
+			continue
+		}
+
+		// CA type: MT4125
+		if strings.HasPrefix(strings.TrimSpace(line), "CA type:") {
+			lines = append(lines, "  "+strings.TrimSpace(line))
+			continue
+		}
+
+		// Number of ports: 1
+		if strings.HasPrefix(strings.TrimSpace(line), "Number of ports:") {
+			lines = append(lines, "  "+strings.TrimSpace(line))
+			continue
+		}
+
+		// Firmware version: 28.39.1002
+		if strings.HasPrefix(strings.TrimSpace(line), "Firmware version:") {
+			lines = append(lines, "  "+strings.TrimSpace(line))
+			continue
+		}
+
+		// Hardware version: 0
+		if strings.HasPrefix(strings.TrimSpace(line), "Hardware version:") {
+			lines = append(lines, "  "+strings.TrimSpace(line))
+			continue
+		}
+
+		// Node GUID: 0xa088c20300e3142a
+		if strings.HasPrefix(strings.TrimSpace(line), "Node GUID:") {
+			lines = append(lines, "  "+strings.TrimSpace(line))
+			continue
+		}
+
+		// System image GUID: 0xa088c20300e3142a
+		if strings.HasPrefix(strings.TrimSpace(line), "System image GUID:") {
+			lines = append(lines, "  "+strings.TrimSpace(line))
+			continue
+		}
+
+		// Port 1:
+		if strings.HasPrefix(strings.TrimSpace(line), "Port 1:") {
```

Comment:
Yeah for now, these are the only values I can fine. We can easily support other formats, if we ever find any :)

Reply:
If we have something like this, it would ignore the "Port 2" line and have duplicates for port 1?  
```
~]$ ibstat mlx4_1
CA 'mlx4_1'
        CA type: MT4099
        Number of ports: 2
        Firmware version: 2.30.8000
        Hardware version: 0
        Node GUID: 0xf4521403007bcba0
        System image GUID: 0xf4521403007bcba3
        Port 1:
                State: Active
                Physical state: LinkUp
                Rate: 56
                Base lid: 2
                LMC: 1
                SM lid: 2
                Capability mask: 0x0251486a
                Port GUID: 0xf4521403007bcba1
                Link layer: InfiniBand
        Port 2:
                State: Active
                Physical state: LinkUp
                Rate: 40
                Base lid: 0
                LMC: 0
                SM lid: 0
                Capability mask: 0x04010000
                Port GUID: 0xf65214fffe7bcba2
                Link layer: Ethernet
```

Reply:
It will be ignored (only catching Port 1). We can add if we have use case for those setups

Reply:
ok, the unmarshal would be wrong. I think that's fine if we have a place that could caught this case.